cli/default.php

```diff
@@ -0,0 +1,112 @@
+
+/**
+ * CLI script for default settings value for activity.
+ *
+ * @package    local_integrity
+ * @copyright  2025 Catalyst IT
+ * @author     Guillaume BARAT (guillaumebarat@catalyst-au.net)
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+use local_integrity\plugininfo\integritystmt;
+use local_integrity\statement_factory;
+
+define('CLI_SCRIPT', true);
+
+require(__DIR__ . '/../../../config.php');
+require_once($CFG->libdir . '/clilib.php');
+
+[$options, $unrecognized] = cli_get_params(
+    [
+        'default' => false,
+        'help' => false,
+    ],
+    [
+        'd' => 'default',
+        'h' => 'help',
+    ]
+);
+
+if ($unrecognized) {
+    $unrecognized = implode("\n  ", $unrecognized);
+    cli_error(get_string('cliunknowoption', 'admin', $unrecognized));
+}
+
+if ($options['help']) {
+    $help = <<<EOT
+Default value for academic integrity.
+
+Options:
+ -d, --default             Set default integrity settings for each activity that has not been set yet.
+
+Example:
+\$sudo -u www-data /usr/bin/php local/integrity/cli/default.php --default=0,1
+
+EOT;
+    cli_writeln($help);
+    exit(0);
+}
+
+if (isset($options['default']) && in_array($options['default'], [0, 1])) {
+    $enabled = $options['default'];
+    $stmt = statement_factory::get_statements();
+    $pluginlist = integritystmt::get_enabled_plugins();
+    foreach ($pluginlist as $name) {
+        $pluginstmt[$name] = $stmt[$name]->get_plugin_name();
+    }
+    [$insqlplugin, $paramsplugins] = $DB->get_in_or_equal(array_keys($pluginlist));
+    $courseids = $DB->get_records('course', null, 'id', 'id');
+    foreach ($courseids as $courseid) {
+        $coursecontext = context_course::instance($courseid->id, IGNORE_MISSING);
+        if (!empty($coursecontext)) {
+            $children = $coursecontext->get_child_contexts();
+            if (!empty($children)) {
+                [$insql, $params] = $DB->get_in_or_equal(array_keys($children));
+                $sql = "SELECT ct.id contextid, m.name modulename
+                            FROM {modules} m
+                            INNER JOIN {course_modules} cm ON m.id = cm.module
+                            INNER JOIN {context} ct ON ct.instanceid = cm.id
+                            WHERE ct.contextlevel = 70 AND cm.course = $courseid->id AND m.name $insqlplugin
+                            ";
+                $check = $DB->get_records_select(\local_integrity\settings::TABLE, "contextid $insql", $params);
+                $contextalreadysettup = [];
+                $datacontextplugins = $DB->get_records_sql($sql, $paramsplugins);
+                foreach ($check as $activity) {
+                    $contextalreadysettup[] = $activity->contextid;
+                }
+                $buildobject = [];
+                foreach ($datacontextplugins as $datacontext) {
+                    if (!in_array($datacontext->contextid, $contextalreadysettup)) {
```

Comment:
Instead of doing in PHP we can do this as part of initial SQL. Basically we need to get ALL records from context table that are not in local_integrity\settings::TABLE table.